### PR TITLE
Implement transports feature

### PR DESCRIPTION
**Pull request by SleepWalker**
This an implementation of transports feature as discussed in #68 

I've also fixed a bug, where you have created facade around non-function properties of logger

### CODE REVIEW

File: test/common/transports.test.js [hunk 1]
```diff
@@ -0,0 +1,78 @@
+/* eslint-env jest */
+
+const logdown = require('../../src/node')
+
+describe('logdown.transports', () => {
+  let initialDebug
+
+  beforeEach(() => {
+    logdown.transports.length = 0 // empty an array
```
Comment (caiogondim) [inline test/common/transports.test.js]: Why not `transports = []`?

**Comment by caiogondim** [comment]
That's is great!
Only a small comment on the PR.

**Comment by caiogondim** [comment]
Should we make `console` itself a transport?
That would make the code more elegant, IMHO.

**Comment by caiogondim** [comment]
I added you as collaborator on the project.
You now have commit access 🎉

**Comment by caiogondim** [comment]
We need to update the README as well

File: test/common/transports.test.js (hunk 1 above)
Comment (SleepWalker) [reply test/common/transports.test.js]: actually, that was for fun :D
I'll change back to `[]`

**Comment by SleepWalker** [comment]
> Should we make console itself a transport?
> That would make the code more elegant, IMHO.

I've had the same thoughts too, but this will complicate transport management. The client will be forced to always keep in mind, that there is a `console` transport, which shouldn't be touched.

E.g you won't be able to do such things:

```js
logdown.transports = [];
logdown.transports = [fooTransport];
```

instead you will be forced to do something like this:

```js
logdown.transports = logdown.transports.slice(0, 1);
logdown.transports = logdown.transports.slice(0, 1).concat(fooTransport);
// or
logdown.transports = [logdown.transports[0]];
logdown.transports = [logdown.transports[0], fooTransport];
// or
logdown.transports.length = 1;
```

If someone will add his transport before `console`, then the code above won't work at all.

As a trade-off we can implement an internal transports array, which will be concatenated with the public one.

> I added you as collaborator on the project.

Great :tada: 

> We need to update the README as well

ok. I'll do that after we will finalize the implementation

**Comment by caiogondim** [comment]
Lets not migrate `console` to `transports`.
That would make `opts.logger` confusing.

**Comment by caiogondim** [comment]
Once we have it merged will give you access to npm so you can publish it 🎉

**Comment by caiogondim** [comment]
Now we need to push a new release.
This is a minor version, since a feature was added and no backward compatibility was broken.

What's your npm username?

**Pull request merged**